GoMeet/src/main/webapp/js/gcal-time-proposal.js

```diff
@@ -0,0 +1,152 @@
+ * if the user entered times where the 'period-end' is earlier than 'period-start', 
+ * or if any one of the times is in the past.
+ * Otherwise, if both 'period-start and 'period-end' are valid, calls the 
+ * GoogleCalendarTimesDAO to fetch the automatically generated meeting times, and 
+ * displays the times on the HTML page.
+ */
+async function generateTimes() {
+  // Verify that both start and end are in the future, and
+  // that end is later than start by at least 1 minute.
+  let startInput = document.getElementById('period-start');
+  let endInput = document.getElementById('period-end');
+  if (startInput.value !== '' && endInput.value !== '' &&
+      rectifyTimeBoundaries(startInput.value, endInput.value)) {
+    // Call the DAO function to fetch times from the servlet and display
+    let generatedTimes = displayAutoGeneratedTimes(
+      await GoogleCalendarTimesDAO.findTimes(startInput.value, endInput.value)
+    );
+    if (generatedTimes !== null) {
+      // Setup the 'Next' button to now allow the user to progress
+      setupNextButton(generatedTimes);
+    } // Otherwise, the error message would be displayed.
+  } else {
+    // Clear both inputs and alert the user 
+    startInput.value = '';
+    endInput.value = '';
+    alert(INVALID_TIME_PERIOD);
+  }
+}
+
+/**
+ * Returns true or false whether start and end times are in the future, 
+ * and whether end is later than start.
+ * @param {String} start The start time, in the format yyyy-MM-ddTHH:mm.
+ * @param {String} end The end time, in the format yyyy-MM-ddTHH:mm
+ * @return True if both start and end are times in the future from now, 
+ * AEDT, and if end is later than start. Otherwise returns false.
+ */
+function rectifyTimeBoundaries(start, end) {
+  // Verify that both start and end are in the future, and
+  // that end is later than start by at least 1 minute.
+  let startDate = new Date(start);
+  let endDate = new Date(end);
+  if (TimeProposalUtil.verifyFutureTime(start) &&
+      TimeProposalUtil.verifyFutureTime(end) &&
+      endDate > startDate) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Displays the server response to the HTML page, in the
+ * 'generated-times' div. 
+ * Will display the error message and button to redirect to the 
+ * manual time proposal page, if serverResponse is an error response, 
+ * otherwise will display each of the times as returned by the server.
+ * @param {Object} serverResponse The response from the 'gcal-find-times' 
+ * server.
+ * @returns {List[String]} The list of times displayed to the user,
+ * or null if the serverResponse was an error response.
+ */
+function displayAutoGeneratedTimes(serverResponse) {
+  let displayDiv = document.getElementById('generated-times');
+  // Clear whatever is in the <div>
+  displayDiv.innerHTML = '';
+
+  if ('status' in serverResponse && parseInt(serverResponse.status) !== 200) {
```

Comment:
Optional: I feel like this could use two helper functions. One to display error message and the other to display the times.

Reply:
I've refactored the error message display to a separate function, because that is more readable 😊 But I've left the code to display the actual times in this function, because given the name of this function `displayAutoGeneratedTimes` I think that code is rightfully left there